File: documentation-examples-java/src/java/dev/typr/foundations/docs/duckdb/MapTypes.java
package dev.typr.foundations.docs.duckdb;

import dev.typr.foundations.DuckDbType;
import dev.typr.foundations.DuckDbTypes;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.Map;
import java.util.UUID;

@SuppressWarnings("unused")
public class MapTypes {
    //start
    // Create map types using the mapTo() combinator
    DuckDbType<Map<String, Integer>> mapStrInt = DuckDbTypes.varchar.mapTo(DuckDbTypes.integer);
    DuckDbType<Map<String, String>> mapStrStr = DuckDbTypes.varchar.mapTo(DuckDbTypes.varchar);
    DuckDbType<Map<UUID, LocalTime>> mapUuidTime = DuckDbTypes.uuid.mapTo(DuckDbTypes.time);

    // Works with any combination of types
    DuckDbType<Map<String, LocalDate>> mapStrDate = DuckDbTypes.varchar.mapTo(DuckDbTypes.date);
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/duckdb/MapTypes.scala
package dev.typr.foundations.docs.duckdb
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*


import java.time.{LocalDate, LocalTime}
import java.util.{Map, UUID}

@SuppressWarnings(Array("unused"))
object MapTypes:
  //start
  // Create map types using the mapTo() combinator
  val mapStrInt: DuckDbType[Map[String, Int]] = DuckDbTypes.varchar.mapTo(DuckDbTypes.integer)
  val mapStrStr: DuckDbType[Map[String, String]] = DuckDbTypes.varchar.mapTo(DuckDbTypes.varchar)
  val mapUuidTime: DuckDbType[Map[UUID, LocalTime]] = DuckDbTypes.uuid.mapTo(DuckDbTypes.time)

  // Works with any combination of types
  val mapStrDate: DuckDbType[Map[String, LocalDate]] = DuckDbTypes.varchar.mapTo(DuckDbTypes.date)
  //stop
